### PR TITLE
Agat recipe

### PR DESCRIPTION
**Pull request by Juke34**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/agat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  license_family: GPL
+  license_file: LICENSE
+  summary: Another Gff Analysis Toolkit (AGAT). Suite of tools to handle gene annotations in any GTF/GFF format.
+  dev_url: https://github.com/NBISweden/AGAT
+
+extra:
+  recipe-maintainers:
+    - juke34
+  skip-lints:
+    - missing_tests
```
Comment (dpryan79) [inline recipes/agat/meta.yaml]: Can you add some tests? There appear to be scripts that one could run, maybe checking the output of `--help` for one of them.

**Comment by Juke34** [comment]
I tried but it didn't work...
I had 

```
test:
     commands:
        - agat_sp_gxf_to_gff3.pl -h
```

but it was crashing at the singularity check.... telling that Bioperl was missing ('Can't locate Bio/Tools/GFF.pm'):

```
.22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] INFO Invoke Task [singularity]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] WARN Invoked Stepless Task [singularity]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] INFO Invoke Task [cleanup]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] STEP Run image [quay.io/dpryan79/mulled_container:latest] with command [[rm -rf /data/dist]]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Creating container [step-ea14abe96d]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Created container [b567ae4fba44 step-ea14abe96d], starting it
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Container [b567ae4fba44 step-ea14abe96d] started, waiting for completion
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [b567ae4fba44 step-ea14abe96d] completed with exit code [0] as expected
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [b567ae4fba44 step-ea14abe96d] removed
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] INFO Invoke Task [test]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] STEP Run image [quay.io/biocontainers/agat:0.0.1--pl526r35_0] with command [[agat_sp_gxf_to_gff3.pl -h]]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Creating container [step-a1d1d5a17b]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Created container [8f77470d14df step-a1d1d5a17b], starting it
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [8f77470d14df step-a1d1d5a17b] started, waiting for completion
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] SERR Can't locate Bio/Tools/GFF.pm in @INC (you may need to install the Bio::Tools::GFF module) (@INC contains: /usr/local/lib/site_perl/5.26.2/x86_64-linux-thread-multi /usr/local/lib/site_perl/5.26.2 /usr/local/lib/5.26.2/x86_64-linux-thread-multi /usr/local/lib/5.26.2 .) at /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm line 10.
```

But le perl package contains tests that are run during installation so I though it was useless to add a test:
```
.08:10:21 BIOCONDA INFO (OUT) t/gff_syntax.t ... ok
08:10:22 BIOCONDA INFO (OUT) t/gff_version.t .. ok
```

File: recipes/agat/meta.yaml (hunk 1 above)
Comment (Juke34) [reply recipes/agat/meta.yaml]: I tried but it didn't work...
I had 

```
test:
     commands:
        - agat_sp_gxf_to_gff3.pl -h
```

but it was crashing at the singularity check (`Can't locate Bio/Tools/GFF.pm`).... telling that Bioperl module was missing while it is installed (`bioconda/linux-64::perl-bioperl-1.6.924-4`). Could it be related to Bioperl version?:

```
.22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] INFO Invoke Task [singularity]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] WARN Invoked Stepless Task [singularity]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] INFO Invoke Task [cleanup]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] STEP Run image [quay.io/dpryan79/mulled_container:latest] with command [[rm -rf /data/dist]]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Creating container [step-ea14abe96d]
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Created container [b567ae4fba44 step-ea14abe96d], starting it
22:01:13 BIOCONDA INFO (ERR) [Nov 20 22:01:13] DEBU Container [b567ae4fba44 step-ea14abe96d] started, waiting for completion
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [b567ae4fba44 step-ea14abe96d] completed with exit code [0] as expected
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [b567ae4fba44 step-ea14abe96d] removed
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] INFO Invoke Task [test]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] STEP Run image [quay.io/biocontainers/agat:0.0.1--pl526r35_0] with command [[agat_sp_gxf_to_gff3.pl -h]]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Creating container [step-a1d1d5a17b]
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Created container [8f77470d14df step-a1d1d5a17b], starting it
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] DEBU Container [8f77470d14df step-a1d1d5a17b] started, waiting for completion
22:01:14 BIOCONDA INFO (ERR) [Nov 20 22:01:14] SERR Can't locate Bio/Tools/GFF.pm in @INC (you may need to install the Bio::Tools::GFF module) (@INC contains: /usr/local/lib/site_perl/5.26.2/x86_64-linux-thread-multi /usr/local/lib/site_perl/5.26.2 /usr/local/lib/5.26.2/x86_64-linux-thread-multi /usr/local/lib/5.26.2 .) at /usr/local/lib/site_perl/5.26.2/AGAT/OmniscientI.pm line 10.
```

But le perl package contains tests that are run during installation so I though it was useless to add a test:
```
.08:10:21 BIOCONDA INFO (OUT) t/gff_syntax.t ... ok
08:10:22 BIOCONDA INFO (OUT) t/gff_version.t .. ok
```

**Comment by dpryan79** [comment]
For some reason the perl include path isn't getting modified when the package is built and that's causing the tests to fail. That suggests that something is going wrong with installation. Maybe @bioconda/perl has a clue why.

**Comment by dpryan79** [comment]
Ah, that version of bioperl was built for an older perl version, it should never have gotten pulled in!

**Comment by Juke34** [comment]
Great thanks it works well now!
Could you add the `please review & merge` label or even better merge the PR?

**Comment by dpryan79** [comment]
I'll merge this once the CI testing is finished :)

**Pull request merged**